src/ui/public/vislib/lib/axis/axis.js

```diff
@@ -43,125 +43,21 @@ export default function AxisFactory(Private) {
 
       const stackedMode = ['normal', 'grouped'].includes(this.axisConfig.get('scale.mode'));
       if (stackedMode) {
-        this.stack.out((d, y0, y) => {
-          return this._stackNegAndPosVals(d, y0, y);
-        });
+        this.stack = this._stackNegAndPosVals;
```

Comment:
I know this is against 5.2, but in hindsight, the implementation remains a little convoluted on master too.

https://github.com/thomasneirynck/kibana/blob/master/src/ui/public/vislib/lib/axis/axis.js#L32-L50

Why do we need to create this  .stack implementation dynamically? It'd just move `.stack` to the prototype, and do a simple switch to trigger the correct code-path there. One for the default d3-based -path, the other for _stackNegAndPosVals.